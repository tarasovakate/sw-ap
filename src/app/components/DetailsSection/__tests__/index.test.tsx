import * as React from 'react';
import { render } from '@testing-library/react';

import { DetailsSection } from '..';

const renderDetailsSection = (
  props: Parameters<typeof DetailsSection>[number],
) => render(<DetailsSection {...props} />);

const mockProps = {
  name: 'Luke',
  birth_year: '41.9BBY',
  gender: 'Male',
  listOfFilms: ['Film 1', 'Film 2'],
  isFetchingFilms: false,
};

describe('<DetailsSection  />', () => {
  it('should match snapshot', () => {
    const detailsSectionComponent = renderDetailsSection({
      ...mockProps,
    });
    expect(detailsSectionComponent.container.firstChild).toMatchSnapshot();
  });

  it('should show loading symbol when fetching films data from API', () => {
    const detailsSectionComponent = renderDetailsSection({
      ...mockProps,
      isFetchingFilms: true,
    });
    expect(
      detailsSectionComponent.queryAllByTestId('svg-loading'),
    ).toHaveLength(1);
  });
  it('should not show loading symbol when films data is received and show the details instead', () => {
    const detailsSectionComponent = renderDetailsSection({
      ...mockProps,
    });
    expect(
      detailsSectionComponent.queryAllByTestId('svg-loading'),
    ).toHaveLength(0);
    expect(detailsSectionComponent.getAllByText(mockProps.name)).toHaveLength(
      1,
    );
    expect(
      detailsSectionComponent.getAllByText(mockProps.birth_year),
    ).toHaveLength(1);
    expect(detailsSectionComponent.getAllByText(mockProps.gender)).toHaveLength(
      1,
    );
    expect(
      detailsSectionComponent.getAllByText(mockProps.listOfFilms.join(', ')),
    ).toHaveLength(1);
  });
});
