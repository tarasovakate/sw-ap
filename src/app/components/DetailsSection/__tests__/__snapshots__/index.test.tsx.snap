// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DetailsSection  /> should match snapshot 1`] = `
.c0 {
  display: grid;
  grid-template-columns: 8rem 1.25rem 1fr;
}

.c1 {
  font-weight: 600;
}

<div
  class=""
>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      Name
    </div>
    <div>
      :
    </div>
    <div
      class=""
    >
      Luke
    </div>
  </div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      Birth year
    </div>
    <div>
      :
    </div>
    <div
      class=""
    >
      41.9BBY
    </div>
  </div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      Gender
    </div>
    <div>
      :
    </div>
    <div
      class=""
    >
      Male
    </div>
  </div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      List of films
    </div>
    <div>
      :
    </div>
    <div
      class=""
    >
      Film 1, Film 2
    </div>
  </div>
</div>
`;
